ygen/enumgen.go

```diff
@@ -744,23 +747,36 @@ func (s *enumGenState) resolveNameClashSet(nameClashSets map[string]map[string]*
 	return uniqueNamesMap, nil
 }
 
-// resolveEnumeratedUnionEntry takes an input YANG union yang.Entry and computes the enumerated
-// values that should be generated for the entry. The name of the enumerated
-// value is calculated based on the original context, whether path compression is enabled based
-// on the compressPaths boolean, and whether the name should not include underscores, as per the
-// noUnderscores boolean.
+// resolveEnumeratedUnionEntry takes an input YANG union yang.Entry and
+// computes the names of the enumerated values that should be generated for the
+// entry. The name of each enumerated value is calculated based on the original
+// context: whether path compression is enabled, whether the name should not
+// include underscores, whether the enum name should be de-deduped if the same
+// enumeration is already been seen in a different part of the YANG tree,
+// whether a shortened version of the enumeration names should be given for
+// enumeration leaves, and whether the defining module name (as opposed to the
+// user module's name) should be used to name enumerated typedefs.
 func (s *enumGenState) resolveEnumeratedUnionEntry(e *yang.Entry, compressPaths, noUnderscores, skipEnumDedup, shortenEnumLeafNames, useDefiningModuleForTypedefEnumNames bool) error {
-	for _, t := range util.EnumeratedUnionTypes(e.Type.Type) {
+	enumTypes := util.EnumeratedUnionTypes(e.Type.Type)
```

Comment:
I think I have the same question here as elsewhere, why is an entry at index `i` of `enumTypes` and `subsumingTypes` equivalent?

Reply:
reverted now that `DefiningType` is being used.